File: dcc/python/2144-minimum-cost-of-buying-candies-with-discount/minimum-cost-of-buying-candies-with-discount.py
class Solution(object):
    def minimumCost(self, cost):
        """
        :type cost: List[int]
        :rtype: int
        """
        maxheap = [-c for c in cost]
        heapq.heapify(maxheap)

        total = 0

        while maxheap:
            total += -heapq.heappop(maxheap)

            if maxheap:
                total += -heapq.heappop(maxheap)

            if maxheap:
                heapq.heappop(maxheap)

        return total

File: cpp/2144-minimum-cost-of-buying-candies-with-discount/minimum-cost-of-buying-candies-with-discount.cpp
class Solution {
public:
    int minimumCost(vector<int>& cost) {
        priority_queue<int> maxheap;

        for (int x : cost) {
            maxheap.push(x);
        }

        int total = 0;
        while (!maxheap.empty()) {
            total += maxheap.top();
            maxheap.pop();

            if (!maxheap.empty()) {
                total += maxheap.top();
                maxheap.pop();
            }

            if (!maxheap.empty()) {
                maxheap.pop();
            }
        }
        return total;
    }
};